### PR TITLE
Add docs to field type validators

### PR DESCRIPTION
**Pull request by dew326**

### CODE REVIEW

File: docs/field-type-validator-usage.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
+# Field Type Validator
+
+Field Type Validator is a base class to validate inputs in field type.
+
+## How to use it?
+
+```javascript
+class Validator extends global.eZ.BaseFieldValidator {
```
Comment (sunpietro) [inline docs/field-type-validator-usage.md]: Pointing to `global` object might be a bit confusing unless you provide a full example:
```javascript
(function (global) {
    class Validator extends global.eZ.BaseFieldValidator { ... }
})(window);
```

File: docs/field-type-validator-usage.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+        return {
+            isError: {Boolean},
+            errorMessage: {String},
+        };
+    }
+};
+
+const validator = new Validator({
+    classInvalid: _{String}_,
```
Comment (sunpietro) [inline docs/field-type-validator-usage.md]: Inside the code block italic formatting won't work.

File: docs/field-type-validator-usage.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+});
+
+validator.init();
+```
+
+## Props list
+
+The `BaseFieldValidator` class has a few required properties. All of them are listed below.
+
+**classInvalid** _{String}_ - CSS class to be added when field is invalid, example: 'is-invalid'.
```
Comment (sunpietro) [inline docs/field-type-validator-usage.md]: to be more specific: `CSS class name`

**Comment by lserwatka** [comment]
@DominikaK documentation should be placed under "documentation" or "docs"?

File: docs/field-type-validator-usage.md [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+**fieldSelector** _{String}_ - selector of the field, example: '.ez-field-edit-ezstring',
```
Comment (sunpietro) [inline docs/field-type-validator-usage.md]: `CSS selector`

Comment (sunpietro) [reply docs/field-type-validator-usage.md]: It can be corrected here and there.

File: docs/field-type-validator-usage.md [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+**eventsMap** _{Array of Objects}_  - events config:
+- **selector** _{String}_ - selector of input (where add event listener), example: '.ez-field-edit-ezstring input'.
+- **eventName** _{String}_ - event name, example: 'blur'.
+- **callback** _{String}_ - callback for event listener, example: 'validateInput', should return object with two params:
+    - **isError** _{Boolean}_ - indicator of error state (true/false).
+    - **errorMessage** _{String}_ - text of the error.
+- **invalidStateSelectors** {Array of Strings} - selectors where add invalid class, example: ['.ez-field-edit-ezstring'].
+- **errorNodeSelectors** {Array of Strings} - selectors where append error message, example: ['.ez-field-edit-text-zone'].
+
+## Useful methods
```
Comment (sunpietro) [inline docs/field-type-validator-usage.md]: I'm missing description why they are useful

File: src/bundle/Resources/public/js/scripts/fieldType/base/base-field.js [hunk 6]
```diff
@@ -8,18 +8,55 @@
...
+         * @param {String} selector
+         * @param {Function} callback
+         * @memberof BaseFieldValidator
+         */
         attachEvent(eventName, selector, callback) {
             [...doc.querySelectorAll(selector)].forEach(item => item.addEventListener(eventName, callback, false));
         }
 
+        /**
+         * Removes event to given selector
```
Comment (sunpietro) [inline src/bundle/Resources/public/js/scripts/fieldType/base/base-field.js]: `from a node found by a given selector`

**Comment by DominikaK** [comment]
I intend to move it to the `developer-documentation` repo anyway, so there's not preference from my side.

File: docs/field-type-validator-usage.md [hunk 7]
```diff
@@ -0,0 +1,73 @@
+# Field Type Validator
+
+Field Type Validator is a base class to validate inputs in field type.
```
Comment (DominikaK) [inline docs/field-type-validator-usage.md]: `Field Type Validator is a base class to validate inputs in a Field Type.`

File: src/bundle/Resources/public/js/scripts/fieldType/base/base-field.js [hunk 8]
```diff
@@ -8,18 +8,55 @@
...
+         * @param {String} selector
+         * @param {Function} callback
+         * @memberof BaseFieldValidator
+         */
         removeEvent(eventName, selector, callback) {
             [...doc.querySelectorAll(selector)].forEach(item => item.removeEventListener(eventName, callback, false));
         }
 
+        /**
+         * Find the nodes to add validation state
```
Comment (sunpietro) [inline src/bundle/Resources/public/js/scripts/fieldType/base/base-field.js]: `Finds nodes`

File: src/bundle/Resources/public/js/scripts/fieldType/base/base-field.js [hunk 9]
```diff
@@ -40,14 +85,43 @@
             return errorNode;
         }
 
+        /**
+         * Find the error containers
```
Comment (sunpietro) [inline src/bundle/Resources/public/js/scripts/fieldType/base/base-field.js]: Would it be a good idea to point out methods that should be overriden when extending a class?

File: src/bundle/Resources/public/js/scripts/fieldType/base/base-field.js [hunk 10]
```diff
@@ -63,6 +137,14 @@
             }
         }
 
+        /**
+         * Validates the field
+         *
+         * @method validateField
+         * @param {Object} config
+         * @param {EventFacade} event
```
Comment (sunpietro) [inline src/bundle/Resources/public/js/scripts/fieldType/base/base-field.js]: just `{Event}`

File: docs/field-type-validator-usage.md [hunk 11]
```diff
@@ -0,0 +1,73 @@
...
+## Props list
+
+The `BaseFieldValidator` class has a few required properties. All of them are listed below.
+
+**classInvalid** _{String}_ - CSS class to be added when field is invalid, example: 'is-invalid'.
+
+**fieldSelector** _{String}_ - selector of the field, example: '.ez-field-edit-ezstring',
+
+**eventsMap** _{Array}_  - events config:
+- **selector** _{String}_ - selector of input (where add event listener), example: '.ez-field-edit-ezstring input'.
```
Comment (DominikaK) [inline docs/field-type-validator-usage.md]: `- **selector** _{String}_ - selector of input (where to add an event listener), example: '.ez-field-edit-ezstring input'.`

File: docs/field-type-validator-usage.md [hunk 12]
```diff
@@ -0,0 +1,73 @@
...
+- **eventName** _{String}_ - event name, example: 'blur'.
+- **callback** _{String}_ - callback for event listener, example: 'validateInput', should return object with two params:
+    - **isError** _{Boolean}_ - indicator of error state (true/false).
+    - **errorMessage** _{String}_ - text of the error.
+- **invalidStateSelectors** {Array} - selectors where add invalid class, example: ['.ez-field-edit-ezstring'].
```
Comment (DominikaK) [inline docs/field-type-validator-usage.md]: `- **invalidStateSelectors** {Array} - selectors where to add invalid class, example: ['.ez-field-edit-ezstring'].`

File: docs/field-type-validator-usage.md [hunk 13]
```diff
@@ -0,0 +1,73 @@
...
+- **errorNodeSelectors** {Array} - selectors where append error message, example: ['.ez-field-edit-text-zone'].
```
Comment (DominikaK) [inline docs/field-type-validator-usage.md]: `- **errorNodeSelectors** {Array} - selectors where to append an error message, example: ['.ez-field-edit-text-zone'].`

File: docs/field-type-validator-usage.md [hunk 14]
```diff
@@ -0,0 +1,73 @@
...
+
+## Useful methods
+
+**init** - a method for adding event listeners
+**reinit** - a method for remove event listeners and adding new ones (useful when DOM has changed)
```
Comment (DominikaK) [inline docs/field-type-validator-usage.md]: `**reinit** - a method for removing event listeners and adding new ones (useful when DOM has changed)`

File: src/bundle/Resources/public/js/scripts/fieldType/ezauthor.js [hunk 15]
```diff
@@ -7,6 +7,14 @@
     const SELECTOR_FIELD_NAME = '.ez-sub-field-name';
 
     class EzAuthorValidator extends global.eZ.BaseFieldValidator {
+        /**
+         * Validates the input name
```
Comment (sunpietro) [inline src/bundle/Resources/public/js/scripts/fieldType/ezauthor.js]: Validates the `name` input field value

File: src/bundle/Resources/public/js/scripts/fieldType/ezauthor.js [hunk 16]
```diff
@@ -18,6 +26,14 @@
             };
         }
 
+        /**
+         * Validates the input email
```
Comment (sunpietro) [inline src/bundle/Resources/public/js/scripts/fieldType/ezauthor.js]: Validates the `email` input field value

**Pull request merged**